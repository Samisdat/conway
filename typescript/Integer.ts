export class Integer {

    public readonly value: number;

    constructor(value: number) {

        this.ensureInteger(value);

        this.value = value;

    }

    private ensureInteger(value: number) {

        if (0 !== value % 1) {
            throw new Error('not an integer value');
        }

    }

}