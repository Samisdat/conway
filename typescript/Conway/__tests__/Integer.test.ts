import {Integer} from '../Integer';

describe('Integer', () => {

    it('can be created', () => {

        let integer = new Integer(1);

        expect(integer).toBeInstanceOf(Integer);

    });

    it('value can be retrieved', () => {

        let integer = new Integer(1);

        expect(integer.value).toBe(1);

    });

    it('throw execption for rows with decimal value', () => {

        expect(() => {
            new Integer(0.5);
        }).toThrowErrorMatchingSnapshot();

    });

});
