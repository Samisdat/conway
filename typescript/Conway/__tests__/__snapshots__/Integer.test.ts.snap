// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integer throw execption for rows with decimal value 1`] = `"not an integer value"`;
